Backports for v0.16.1

lostella opened the pull request:
*Description of changes:* backporting changes
- #3251 
- #3252
- #3254 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.


**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

shchur committed:
Allow disabling scaler for DL models (#3251)

*Issue #, if available:*

*Description of changes:*
- Currently, the incorrect type hint makes it impossible to set
`scaling=None` for TiDE, PatchTST, DLinear and LagTST models. If user
sets `scaling=None`, Pydantic will raise the ValidationError. This PR
fixes this issue.


By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.


**Please tag this pr with at least one of these labels to make our
release process faster:** BREAKING, new feature, bug fix, other change,
dev setup

lostella committed:
Upgrade docs dependencies (#3252)

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.


**Please tag this pr with at least one of these labels to make our
release process faster:** BREAKING, new feature, bug fix, other change,
dev setup

shchur commented:
One more minor fix for a FutureWarning: #3254. Do you think we can squeeze that into 0.16.1?

shchur committed:
Set `observed=True` inside `DataFrame.groupby` in `PandasDataset` to silence the FutureWarning (#3254)

*Issue #, if available:*

*Description of changes:*
- Set `observed=True` inside `DataFrame.groupby` in `PandasDataset` to
silence the FutureWarning. Relevant pandas documentation:
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html#pandas-dataframe-groupby.


By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.


**Please tag this pr with at least one of these labels to make our
release process faster:** BREAKING, new feature, bug fix, other change,
dev setup

lostella commented:
@shchur added

shchur reviewed (approved):
Thanks!
